src/intersectionWith.js

```diff
@@ -12,7 +12,7 @@ var uniqWith = require('./uniqWith');
  * @memberOf R
  * @since v0.1.0
  * @category Relation
- * @sig (a -> a -> Boolean) -> [*] -> [*] -> [*]
+ * @sig ((a, a) -> Boolean) -> [a] -> [a] -> [a]
```

Comment:
interesting. I see this as a "set" operation, where the user is saying these two sets are the same type based on the predicate. I'm not gonna make a stand here, but i would expect an intersection of `[a]` and `[b]` to be `[ ]`.